workflow/controller/controller.go

```diff
@@ -54,6 +54,7 @@ import (
 )
 
 const enoughTimeForInformerSync = 1 * time.Second
+const defaultMaxStackDepth = 500
```

Comment:
This is probably too high for a default... I'm thinking 100-200?

Reply:
I'm not sure we should make this configurable - what about a sensible default?

Reply:
This is no longer configurable, but protected with an envvar feature flag